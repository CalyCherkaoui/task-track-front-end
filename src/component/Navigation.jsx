/* eslint-disable no-console */
import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { Navbar, Nav } from 'react-bootstrap';
import styles from '../styles/Navigation.module.css';
import { logout } from '../actions/authentication';

const Navigation = () => {
  const { isLoggedIn } = useSelector((state) => state.authentication);
  const dispatch = useDispatch();

  const logOut = () => {
    dispatch(logout());
  };

  return (
    <Navbar className={`${styles.navigation_container} justify-content-between`}>
      <Navbar.Brand href="/">
        <span className={styles.nav_logo_typo_main}>TASK</span>
        <span className={styles.nav_logo_typo_second}>
          &Track
        </span>
      </Navbar.Brand>
      <Nav className="">

        {
            isLoggedIn ? (
              <li className="nav-item">
                <Link className={`${styles.nav_link_typo} nav-link`} to="/profile">
                  Profile
                </Link>
              </li>
            ) : (
              <li className="nav-item">
                <Link className={`${styles.nav_link_typo} nav-link`} to="/about">About</Link>
              </li>
            )
        }
        {
            isLoggedIn ? (
              <li className="nav-item">
                <a href="/login" className={`${styles.nav_link_typo} nav-link`} onClick={logOut}>
                  LogOut
                </a>
              </li>
            ) : (
              <div className="navbar-nav ml-auto">
                <li className="nav-item">
                  <Link to="/login" className={`${styles.nav_link_typo} nav-link`}>
                    Login
                  </Link>
                </li>
                <li className="nav-item">
                  <Link to="/signup" className={`${styles.nav_link_typo} nav-link`}>
                    Sign-Up
                  </Link>
                </li>
              </div>
            )
          }
      </Nav>
    </Navbar>
  );
};

export default Navigation;
