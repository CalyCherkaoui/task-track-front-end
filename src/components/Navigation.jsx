import React from 'react';
import { Link } from 'react-router-dom';
import { IconContext } from 'react-icons';
import { BsCircleFill } from 'react-icons/bs';
import styles from '../styles/Navigation.module.css';

const Navigation = () => (
  <div className={`${styles.navigation_bar} flex_row flex_space_between shadowed_small`}>
    <div className={`${styles.navigation_left} flex_row flex_start_bl`}>
      <p className={`${styles.nav_logo} bold_typography`}>
        Trackmy-habits|
        <span className={styles.nav_logo_emph}>
          <IconContext.Provider value={{ className: 'logo_icon' }}>
            <BsCircleFill />
          </IconContext.Provider>
        </span>
      </p>
      <ul className="flex_row flex_start_bl">
        <li className={styles.nav_item}>
          <Link to="/" className={`${styles.nav_link} spaced_typography`}>Home</Link>
        </li>
        <li className={styles.nav_item}>
          <Link to="/about" className={`${styles.nav_link} spaced_typography`}>About</Link>
        </li>
        <li className={styles.nav_item}>
          <Link to="/login" className={`${styles.nav_link} spaced_typography`}>Login</Link>
        </li>
        <li className={styles.nav_item}>
          <Link to="/signup" className={`${styles.nav_link} spaced_typography`}>Signup</Link>
        </li>
        <li className={styles.nav_item}>
          <Link to="/profile" className={`${styles.nav_link} spaced_typography`}>Profile</Link>
        </li>
      </ul>
    </div>
  </div>
);

export default Navigation;
