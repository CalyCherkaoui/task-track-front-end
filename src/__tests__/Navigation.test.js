import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store';
import Navigation from '../component/Navigation';

describe(
  'Renders correctly The navigation bar component',
  () => {
    let container;
    beforeEach(() => {
      container = shallow(
        <Provider store={store}>
          <Router>
            <Navigation />
          </Router>
        </Provider>,
      );
    });

    it(
      'Contains About page link',
      () => {
        expect(container.find('About')).toBeTruthy();
      },
    );

    it(
      'Matches Navigation snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <Navigation />
            </Router>
          </Provider>,
        );

        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
