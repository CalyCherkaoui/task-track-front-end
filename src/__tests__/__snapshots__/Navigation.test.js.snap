// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly The navigation bar component Matches Navigation snapshot 1`] = `
<nav
  className="navigation_container justify-content-between navbar navbar-expand navbar-light"
>
  <a
    className="navbar-brand"
    href="/"
  >
    <span
      className="nav_logo_typo_main"
    >
      TASK
    </span>
    <span
      className="nav_logo_typo_second"
    >
      &Track
    </span>
  </a>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  >
    <li
      className="nav-item"
    >
      <a
        className="nav_link_typo nav-link"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </li>
    <div
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav_link_typo nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav_link_typo nav-link"
          href="/signup"
          onClick={[Function]}
        >
          Sign-Up
        </a>
      </li>
    </div>
  </div>
</nav>
`;
